### PR TITLE
Add support for usage of signed JWT for authentication

### PR DESCRIPTION
**Pull request by fraxachun**
Admin:
- ImageBlocks: Support basePath of api (e.g. /dam) when creating preview-urls
- Create alternative useUser-Hook which get’s data from the API
- Provide an alternative for the Admin-Header User Item which uses the new useUser-Hook
- Webs have to avoid using components which use react-app-auth

Api:
- Auth-Module supports Jwt-Strategy but is still backwards-compatible
- Add CometGuard which allows choosing strategies in Web
- Add Resolver for Current-User

Site:
- Avoid errors when next-auth does not exist in Web
- Add validation of JWT which can be used in Web

### CODE REVIEW

File: packages/admin/cms-admin/src/blocks/PixelImageBlock.tsx [hunk 1]
```diff
@@ -38,12 +38,11 @@ export function createPreviewUrl({ damFile, cropArea }: ImageBlockState, apiUrl:
             : [imageCropArea.width, imageCropArea.height, imageCropArea.focalPoint, imageCropArea.x, imageCropArea.y];
     const filename = damFile.name.substr(0, damFile.name.lastIndexOf("."));
 
-    let urlTemplate = urlTemplateRoute;
+    let urlTemplate = apiUrl + urlTemplateRoute;
```
Comment (nsams) [inline packages/admin/cms-admin/src/blocks/PixelImageBlock.tsx:41]: couldn't the /api prefix be part of `urlTemplateRoute`?

File: packages/admin/cms-admin/src/http/createHttpClient.ts [hunk 2]
```diff
@@ -1,28 +1,28 @@
...
         headers,
     });
-    axiosClient.interceptors.request.use(requestInterceptor);
+    if (authorizationManager) {
+        axiosClient.interceptors.request.use(async (config: AxiosRequestConfig) => {
+            if (!authorizationManager.state.oAuth?.accessToken) {
+                return Promise.reject("Can't get user access token");
+            }
+
+            config.headers.Authorization = `Bearer ${authorizationManager.state.oAuth?.accessToken}`;
```
Comment (nsams) [inline packages/admin/cms-admin/src/http/createHttpClient.ts:21]: is this still required?

File: packages/admin/cms-admin/src/index.ts [hunk 3]
```diff
@@ -76,6 +77,7 @@ export type { SiteConfig } from "./sitesConfig/SitesConfigContext";
 export { SitesConfigProvider } from "./sitesConfig/SitesConfigProvider";
 export { useSiteConfig } from "./sitesConfig/useSiteConfig";
 export { useSitesConfig } from "./sitesConfig/useSitesConfig";
+export { useUser } from "./user/useUser";
```
Comment (nsams) [inline packages/admin/cms-admin/src/index.ts:80]: I wouldn't export this as public api as it only loads name+role and can't be extended in any way. I would even say there should not be a useUser as private api in cms-admin - couldn't you make a standard gql request just in the component where you need the user?

File: packages/api/cms-api/src/auth/dto/current-user.dto.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { Field, ObjectType } from "@nestjs/graphql";
+
+@ObjectType()
+export class CurrentUser {
```
Comment (nsams) [inline packages/api/cms-api/src/auth/dto/current-user.dto.ts:4]: We need a way to make this extendable by the application. eg. to add allowed domains

**Review by nsams — COMMENTED** [review]
I would not try to be backwards compatible - that is harder to understand and test.

I would even say - if oauth2-proxy is the way to go (to fix our bugs) - switching to it is required for comet v4.

File: packages/admin/cms-admin/src/common/header/HeaderAboutItem.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import { HeaderAboutButton } from "./about/AboutButton";
+import { HeaderButton, HeaderDropdownContent, HeaderSeparator } from "./Header.sc";
+
+const signOutMutation = gql`
+    mutation SignOut {
+        signOut
+    }
+`;
+
+export function HeaderAboutItem(): React.ReactElement {
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/common/header/HeaderAboutItem.tsx:18]: Is this second component to still be compatible to the "old" `useUser` hook?

File: packages/admin/cms-admin/src/common/header/Header.sc.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import { Box, Button as MUIButton } from "@mui/material";
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/common/header/Header.sc.ts:1]: *.sc.ts files are considered to be "private" files of the respective components and should not be used across multiple components.

**Comment by johnnyomair** [comment]
> I would not try to be backwards compatible - that is harder to understand and test.

I'm not convinced that being backwards compatible is good too. This adds lots of edge cases and duplicate code that is hard to understand.

>I would even say - if oauth2-proxy is the way to go (to fix our bugs) - switching to it is required for comet v4.

If we want to enforce oauth2-proxy, upgrading should be as seamless as possible. We should document this change extensively.
